getting_started/step_by_step/7_room_two.rst

```diff
@@ -189,11 +200,56 @@ when you click the door to leave the pub.
 
 Configure the change between rooms
 ----------------------------------
+
 What we now need is to tell Escoria to move the character to the correct
 location when they change between rooms. For this we will use a script that
 will run when the room first loads. It will check which room the player has
 come from and immediately send them to the correct ``ESCLocation``.
 
+Setup and Ready events
+^^^^^^^^^^^^^^^^^^^^^^
+
+To finish the switch between rooms, we need to explain how the room change
+works. When you activate an ``ESCExit`` (or ``ESCItem`` with "is exit"
+selected), you use the ``change_scene`` command to tell Escoria to replace the
+current room with new room. It doesn't tell the Escoria anything about how to
+set up the new room or where to put the player - this is the responsibility
+of the new room.
+
+Each room's root node is an ``ESCRoom`` node, and the script attached to this
+(if there is one) is responsible for setting up anything dynamic in the room.
+When ``change_scene`` loads the room, it will look in the new room's script for
+a ``setup`` event. Any commands you put in this script will run before the
+player sees the new room - use this to reset any movable objects to their start
+positions.
+
+After the ``setup`` event completes, a transition will run (there is a default
+one set in the Escoria configuration settings), but typically the transition
+will be some sort of nice fade-in effect to show the room. Once the transition
+in runs, Escoria will run any commands in the ``ready`` event. Place any
+commands in here that will tell the next part of your game's story (e.g. make
+the player walk to the window and comment on how it looks like a storm is
+approaching.)
+
+.. hint::
+
+   If you do not have a script attached to your ``ESCRoom`` the player will
+   automatically be placed at a start location called ``player_start`` if you
+   have created an ``ESCLocation`` with this name as part of the room.
+   If an ``ESCLocation`` with this name does not exist, the player will start
+   as close to coordinate (0,0) as possible.
+
+.. hint::
+
+   There can only be one of each type of event per script. If you want multiple
+   steps to run as part of an event (the ``ready`` event for example), place
+   all the commands within the one event block.
+
+For more detail on the order of events in Escoria, see :
```

Comment:
```suggestion
   all the commands within the one event block.

.. hint::

    Transitions are run by default as part of the ``change_scene`` 
    command, but this can be manually overridden if desired.

For more detail on the order of events in Escoria, see :
```